uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/AbsenceTypeTest.php

```diff
@@ -793,4 +798,50 @@ public function testExceptionIsThrownWhenUpdatingAnAbsenceTypeWithTitleOfAnother
 
     AbsenceTypeFabricator::fabricate($params);
   }
+
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidAbsenceTypeException
+   * @expectedExceptionMessage You cannot add public holiday to entitlement when Absence Type calculation unit is in Hours
+   */
+  public function testPublicHolidayEntitlementCannotBeAddedWhenLeaveIsCalculatedInHours() {
+    AbsenceTypeFabricator::fabricate([
+      'add_public_holiday_to_entitlement' => true,
+      'calculation_unit' => $this->calculationUnitOptions['hours']
+    ]);
+  }
+
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidAbsenceTypeException
+   * @expectedExceptionMessage The Calculation unit cannot be change because the Absence Type is In Use!
+   */
+  public function testCalculationUnitCannotBeChangedWhenAbsenceTypeIsInUse() {
+    $absenceType = AbsenceTypeFabricator::fabricate([
+      'calculation_unit' => $this->calculationUnitOptions['hours']
+    ]);
+
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'type_id' => $absenceType->id,
+      'contact_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-01-02'),
+      'to_date' => CRM_Utils_Date::processDate('2016-01-02'),
+    ]);
+
```

Comment:
How about using the setExpectedException() method here instead of the annotation?